sky/packages/sky/lib/src/painting/box_painter.dart

```diff
@@ -425,7 +426,8 @@ void paintImage({
   Size imageSize = new Size(image.width.toDouble(), image.height.toDouble());
   Size sourceSize;
   Size destinationSize;
-  switch(fit) {
+  fit ??= centerSlice == null ? ImageFit.scaleDown : ImageFit.fill;
+  switch (fit) {
```

Comment:
this computes the destinationSize based on the entire image size, not the center size. I think it should compute it based on the center size, with the bounds being the size minus the difference between the image size and the center size.